enhancements/kubernetes-nmstate-integration.md

```diff
@@ -0,0 +1,124 @@
+
+Node-networking configuration driven by Kubernetes and executed by
+[nmstate](https://nmstate.github.io/).
+
+## Motivation
+
+With hybrid clouds, node-networking setup is becoming even more challenging.
+Different payloads have different networking requirements, and not everything
+can be satisfied as overlays on top of the main interface of the node (e.g.
+SR-IOV, L2, other L2).
+The [Container Network Interface](https://github.com/containernetworking/cni)
+(CNI) standard enables different
+solutions for connecting networks on the node with pods. Some of them are
+[part of the standard](https://github.com/containernetworking/plugins), and there are
+others that extend support for [Open vSwitch bridges](https://github.com/kubevirt/ovs-cni),
+[SR-IOV](https://github.com/hustcat/sriov-cni), and more...
+
+However, in all of these cases, the node must have the networks setup before the
+pod is scheduled. Setting up the networks in a dynamic and heterogenous cluster,
+with dynamic networking requirements, is a challenge by itself - and this is
+what this project is addressing.
+
+### Goals
+
+- Deploy Kubnetes-nmstate as part of openshift
+
+### Non-Goals
+
+- Replace SRIOV operator
+
+## Proposal
+
+A new Kubernetes-nmstate handler DaemonSet is deployed in the cluster part of the OpenShift installation.
+This DaemonSet contains nmstate package and interacts with the NetworkManager
+on the host by mounting the related dbus. The project contains two
+Custom Resource Definitions, `NodeNetworkState` and `NodeNetworkConfigurationPolicy`.
+`NodeNetworkState` objects are created per each node in the cluster and can be
+used to report available interfaces and network configuration. These objects
+are created by kubernetes-nmstate and must not be touched by a user.
+`NodeNetworkConfigurationPolicy` objects can be used to specify desired
+networking state per node or set of nodes. It uses API similar to `NodeNetworkState`.
+
+Kubernetes-nmstate DaemonSet creates a custom resource of `NodeNetworkState` type per each node and
+updates the network topology from each OpenShift node.
+
+User configures host network changes and apply a policy in `NodeNetworkConfigurationPolicy` custom
+resource. Network topology is configured via `desiredState` section in `NodeNetworkConfigurationPolicy`.
+Multiple `NodeNetworkConfigurationPolicy` custom resources can be created, Kubenetes-nmstate DaemonSet will merge
+all the `NodeNetworkConfigurationPolicy` custom resources according to priority definition and
+apply one profile that best matches with the node and update the `desiredState` section in `NodeNetworkState`.
+
+Upon receiving a notification event of `NodeNetworkState` update,
+Kubenetes-nmstate Daemon verify the correctness of `NodeNetworkState` custom resource and
+apply the selected profile to the specific node.
+
+### User Stories
+
+#### Bond creation
+
+Be able to create bond interfaces on OpenShift nodes.
```

Comment:
I'd like to see some examples where MCO + Ignition isn't sufficient. I know there are some, but I'd like to see them spelled out.

Reply:
Added one example. Let me know if you'd like more?